CHANGELOG.md

```diff
@@ -0,0 +1,174 @@
+represented by the pull requests that fixed them. This log originated with Singularity 2.4
+and changes prior to that are (unfortunately) done retrospectively. From @olifre, critical items to know are:
+
+ - renamed commands
+ - deprecated / removed commands
+ - changed defaults
+ - backward incompatible changes (recipe file format? image file format?)
+ - migration guidance (how to convert images?)
+ - changed behaviour (recipe sections work differently)
+
+
+## [v3.0](https://github.com/singularityware/singularity/tree/development) (development)
+
+**Implemented enhancements**
+ - This changelog was added.
+
+**Bug Fixes**
+ - .wh.* files leading to permissions errors from Docker aufs extractions are skipped on extraction
+ - The `pull` command respects the user's custom name, and otherwise gives an extension based on file type
+ - Tar creation uses GNU tar-1.28 to not add a timestamp to the header for more reproducible builds
+
+## [v2.4](https://github.com/singularityware/singularity/tree/v2.4) (2017-10-02)
+[Full Changelog](https://github.com/singularityware/singularity/compare/2.3.2...2.4)
+
+**Implemented enhancements**
+
+ - a new `build` command was added to replace `create` + `bootstrap` ([build](https://singularityware.github.io/docs-build-container))
+ - default image format is squashfs, eliminating the need to specify a size
+ - for development build supports `--sandbox` (folder) and `--writable` (ext3)
+ - a `localimage` can be used as a build base, including ext3, sandbox, and other squashfs images
+ - singularity hub can now be used as a base with the uri `shub://`
+ - support has been added for instances (services) including network namespace isolation under the `instances` group of commands.
+ - [singularity registry](https://www.github.com/singularityhub/sregistry) is released and published
+ - [Standard Container Integration Format](https://singularityware.github.io/docs-apps) apps are added to support internal modularity and organization.
+ - [build environment](https://singularityware.github.io/build-environment) is better documented
+ - Persistent Overlay 
+ - Container checks
+ - Tests for instance support
+ - Wrapper for create
+ - Group instance commands
+ - Group image commands
+ - Bash completion updates
+
+**Deprecated**
+ - the `create` command is being deprecated in favor of `image.create`
+ - `bootstrap` is being deprecated in favor of `build` (will work through 2.4)
+ - `expand` is being deprecated, as a container can be used as a build base to change it (will work through 2.4)
+ - `export` is being deprecated and added to the image command group, `image.export`
+ - the `shub://` URI no longer supports an integer to reference a container
+
+## [v2.3.2](https://github.com/singularityware/singularity/tree/v2.3.2) (2017-09-15)
+[Full Changelog](https://github.com/singularityware/singularity/compare/2.3.1...2.3.2)
+
+**Implemented enhancements**
```

Comment:
I would prefer if we used 3 hashes instead of bold to delineate subheadings in keeping with markdown syntax.  And as an added bonus things will work better in the future if we ever decide to add a toc and put it up online or something.

Reply:
aye aye cap-i-tan!

Reply:
https://www.youtube.com/watch?v=46goXXCm9Fo